subsys/testsuite/ztest/Kconfig

```diff
@@ -91,6 +91,31 @@ config ZTEST_RULE_1CPU
 
 endmenu
 
+config ZTEST_DO_THE_SHUFFLE
+	bool "Enables the shuffling of tests and suites"
+	select TEST_RANDOM_GENERATOR if !ENTROPY_HAS_DRIVER
+	default n
+	help
+	  This rule will shuffle the order of tests and test suites.
+
+if ZTEST_DO_THE_SHUFFLE
+config ZTEST_DO_THE_SHUFFLE_SUITE_REPEAT_COUNT
```

Comment:
I believe that this approach would work only for `native_posix` no? Is it not possible to seed/get random values in non `native_posix` platforms? In those platforms it might not be possible to pass command line arguments and having to rely on the caller to re-run actually means having to re-compile with a different seed.

Reply:
If the intention is to use this also for embedded targets, then that's fine. (you can't dismiss this comment in that case)

Reply:
Yes, I believe that the intention is to _also_ use in a hardware lab.